src/query-runner/BaseQueryRunner.ts

```diff
@@ -324,10 +324,18 @@ export abstract class BaseQueryRunner {
                 return false;
         }
 
-        if (this.connection.driver.dataTypeDefaults
+        const defaultWidthForType = this.connection.driver.dataTypeDefaults
             && this.connection.driver.dataTypeDefaults[column.type]
-            && this.connection.driver.dataTypeDefaults[column.type].width) {
-            return this.connection.driver.dataTypeDefaults[column.type].width === width;
+            && this.connection.driver.dataTypeDefaults[column.type].width;
+
+        if (defaultWidthForType) {
+            // the default widths of these numeric types are 1 less when the column is unsigned.
```

Comment:
is this statement true for all databases?

Reply:
No, actually only MariaDB. The method this code lives in (`isDefaultColumnWidth`) is only called for the MySQL driver, as stated in the comment above it, and it turns out that the method is actually only hit when using MariaDB specifically, because [this line](https://github.com/typeorm/typeorm/pull/6632/files/414a3faada18a7574e7dfba98006304cb2df0d36#diff-f449ebcfffaeef0bb22211a613c8e283R1301) always results in `width = undefined` for MySQL due to it's representation of `dbColumn["COLUMN_TYPE"]` not including the width part.

Reply:
Having this check for all drivers is confusing. We need to move this logic to a driver-specific method, e.g. `driver.compareColumnWidths` or something like that.

Reply:
Makes sense, I agree. I can do this refractor as an addition to this PR.

Ps if you are planning to make a release soon let me know so I can do the changes in time.

Reply:
yeah its already quite a while I didn't make a release, so yeah, expect a release soon

Reply:
OK I just looked into this more. The `isDefaultColumnWidth` method is actually used in 2 QueryRunners: MysqlQueryRunner & AuroraQueryRunner. I guess that's why it was put on the base class.

I don't see any tests for Aurora in CI, so I've no way to be sure that my changes are valid for Aurora. 

I question the value of putting it in the Driver interface because as far as I can tell it is an (almost useless) feature of MySQL _only_. [Here's a good explanation](https://alexander.kirk.at/2007/08/24/what-does-size-in-intsize-of-mysql-mean/) of what it actually does.

Here's another suggestion:

1. Move the `isDefaultColumnWidth` method from the Base class to both MysqlQueryRunner and AuroraQueryRunner.
2. Leave the AuroraQueryRunner version as it was before the fix, since I do not know whether it breaks anything there.
3. In the MysqlQueryRunner include the changes from this PR, with an additional check of the existing `isMariaDb` boolean to further make sure (and make explicit) that the fix only applies to MariaDB.

What do you think?

Reply:
yeah, go ahead